src/BuildScriptGenerator/Node/NodePlatform.cs

```diff
@@ -181,9 +182,18 @@ public PlatformDetectorResult Detect(RepositoryContext context)
             if (ctx.SourceRepo.FileExists(NodeConstants.YarnLockFileName))
             {
                 packageManagerCmd = NodeConstants.YarnCommand;
-                packageInstallCommand = NodeConstants.YarnPackageInstallCommand;
                 configureYarnCache = true;
                 packageInstallerVersionCommand = NodeConstants.YarnVersionCommand;
+                // yarn.lock file is valid YAML in Yarn 2+ and yarnrc.yml file replaces yarnrc in Yarn 2+.
```

Comment:
so a repo that doesnt have .yarnrc.yml but has yaml formatted yarn.lock will be considered as yarn2 supported ...

do we commit .yarnrc.yml in the repo too? if not then we may not need to check for this

Reply:
yarn.lock file can be parsed as valid YAML format since yarn 2. The .yarnc.yml file does not have to be in the repo, it's a configuration file for yarn, sometimes it may not be in the repo for yarn2. So I added both as indicators.